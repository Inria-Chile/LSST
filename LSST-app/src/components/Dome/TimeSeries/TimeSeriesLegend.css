.timeseries-legend-element {
    display: inline-block;
    padding: 10px 20px;
}