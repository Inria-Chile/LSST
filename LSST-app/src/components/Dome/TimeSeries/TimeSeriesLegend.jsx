import React, { PureComponent } from 'react'
import './TimeSeriesLegend.css';


class TimeSeriesLegend extends PureComponent {
    static LineColors = {
        'azimuth': '#ddd', 
        'target': '#5e8ba9', 
        'optimal': '#5e8ba9'
    }

    render() {
        return (
            <div className='timeseries-legend'>
                {this.props.legendData.map((element) => {
                    return (
                        <div key={element.label} className='timeseries-legend-element'>
                            <svg height="30" width="40">
                                <line x1="0" y1="15" x2="35" y2="15" style={{stroke: element.color, strokeDasharray: element.lineDash, strokeWidth:7}}/> 
                            </svg> 
                            <span>
                                {element.label}
                            </span>
                        </div>
                    )
                })}
            </div>
        );
    }
}


export default TimeSeriesLegend;