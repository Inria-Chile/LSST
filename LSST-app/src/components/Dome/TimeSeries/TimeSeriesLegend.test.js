import React from 'react';
import TimeSeriesLegend from './TimeSeriesLegend';
import renderer from 'react-test-renderer';
import shallowRenderer from 'react-test-renderer/shallow';

/*---- Snapshot testing ----*/
describe('TimeSeriesLegend_Test', function(){

  it('renders correctly', () => {
    const renderer = new shallowRenderer();
    let legendData = [
        { label: 'Current value', color: '#ddd', lineDash: 0 }, 
        { label: 'Target', color: '#5e8ba9', lineDash: 0 }, 
        { label: 'Optimal', color: '#3e6b89', lineDash: 7 }, 
    ];
    const tree = renderer.render(    
    <TimeSeriesLegend  legendData = {legendData} />
    );
    expect(tree).toMatchSnapshot();
  });
})


