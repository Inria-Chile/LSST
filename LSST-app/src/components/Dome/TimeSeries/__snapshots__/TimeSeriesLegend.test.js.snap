// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSeriesLegend_Test renders correctly 1`] = `
<div
  className="timeseries-legend"
>
  <div
    className="timeseries-legend-element"
  >
    <svg
      height="30"
      width="40"
    >
      <line
        style={
          Object {
            "stroke": "#ddd",
            "strokeDasharray": 0,
            "strokeWidth": 7,
          }
        }
        x1="0"
        x2="35"
        y1="15"
        y2="15"
      />
    </svg>
    <span>
      Current value
    </span>
  </div>
  <div
    className="timeseries-legend-element"
  >
    <svg
      height="30"
      width="40"
    >
      <line
        style={
          Object {
            "stroke": "#5e8ba9",
            "strokeDasharray": 0,
            "strokeWidth": 7,
          }
        }
        x1="0"
        x2="35"
        y1="15"
        y2="15"
      />
    </svg>
    <span>
      Target
    </span>
  </div>
  <div
    className="timeseries-legend-element"
  >
    <svg
      height="30"
      width="40"
    >
      <line
        style={
          Object {
            "stroke": "#3e6b89",
            "strokeDasharray": 7,
            "strokeWidth": 7,
          }
        }
        x1="0"
        x2="35"
        y1="15"
        y2="15"
      />
    </svg>
    <span>
      Optimal
    </span>
  </div>
</div>
`;
